exchanges/huobi/huobi_test.go

```diff
@@ -1839,21 +1839,24 @@ func TestPairFromContractExpiryCode(t *testing.T) {
 			continue // Next Quarter is intermittently present
 		}
 		require.NoErrorf(t, err, "pairFromContractExpiryCode must not error for %s code", cType)
-		assert.Equal(t, currency.BTC, p.Base, "pair Base should be the same")
+		assert.Equal(t, currency.BTC, p.Base, "pair Base should be BTC")
 		h.futureContractCodesMutex.RLock()
-		exp, ok := h.futureContractCodes[cType]
+		cachedContract, ok := h.futureContractCodes[cType]
 		h.futureContractCodesMutex.RUnlock()
 		require.True(t, ok, "%s type must be in contractExpiryNames", cType)
-		assert.Equal(t, currency.BTC, p.Base, "pair Base should be the same")
-		assert.Equal(t, exp, p.Quote, "pair Quote should be the same")
+		assert.Equal(t, cachedContract, p.Quote, "pair Quote should match contractExpiryNames")
 		d, err := time.ParseInLocation("060102", p.Quote.String(), tz)
 		require.NoError(t, err, "currency code must be a parsable date")
-		require.Falsef(t, d.Before(n), "%s expiry must be today or after", cType)
+		require.Falsef(t, d.Before(today), "%s expiry must be today or after", cType)
 		switch cType {
-		case "CW", "NW":
-			require.Truef(t, d.Before(n.AddDate(0, 0, 14)), "%s expiry must be within 14 days; Got: `%s`", cType, d)
-		case "CQ", "NQ":
-			require.Truef(t, d.Before(n.AddDate(0, 6, 0)), "%s expiry must be within 6 months; Got: `%s`", cType, d)
+		case "CW":
+			require.Truef(t, d.Before(today.AddDate(0, 0, 14)), "Current week expiry must be within 14 days; Got: `%s`", d)
```

Comment:
Can we add in ctype again + the unix timestamp today so that if this happens again we can replicate and mock this?

Reply:
How's this?

Fixed gbjk@564c2a65

```
    huobi_test.go:1862:
                Error Trace:    /Users/gbjk/go/src/github.com/gbjk/gocryptotrader/exchanges/huobi/huobi_test.go:1862
                Error:          "15" is not less than or equal to "14"
                Test:           TestPairFromContractExpiryCode
                Messages:       CW expiry must be within expected update window; Today: `2025-05-16`, Expiry: `2025-05-31`
```